packages/mask/src/extension/popups/pages/Personas/VerifyWallet/index.tsx

```diff
@@ -0,0 +1,163 @@
+import { memo, useEffect, useState } from 'react'
+import { useAsync, useAsyncFn, useLocation } from 'react-use'
+import { useNavigate } from 'react-router-dom'
+import { NextIDAction, NextIDPlatform, PopupRoutes } from '@masknet/shared-base'
+import { makeStyles, usePopupCustomSnackbar } from '@masknet/theme'
+import { NextIDProof } from '@masknet/web3-providers'
+import { ChainId, EthereumRpcType, isSameAddress, NetworkType, ProviderType } from '@masknet/web3-shared-evm'
+import type { Web3Plugin } from '@masknet/plugin-infra/dist/web3-types'
+import { SignSteps, Steps } from '../../../../../components/shared/VerifyWallet/Steps'
+import Services from '../../../../service'
+import { PersonaContext } from '../hooks/usePersonaContext'
+import { useTitle } from '../../../hook/useTitle'
+import { useI18N } from '../../../../../utils'
+import { useUnconfirmedRequest } from '../../Wallet/hooks/useUnConfirmedRequest'
+import { PopupContext } from '../../../hook/usePopupContext'
+import urlcat from 'urlcat'
+
+const useStyles = makeStyles()((theme) => ({
+    container: {
+        display: 'flex',
+        flexDirection: 'column',
+        flex: 1,
+        background: 'rgba(247, 249, 250, 1)',
+        padding: '8px 16px 16px 16px',
+    },
+}))
+const VerifyWallet = memo(() => {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const { currentPersona, refreshProofs } = PersonaContext.useContainer()
+    const { signed, setSigned } = PopupContext.useContainer()
+    const [isBound, setIsBound] = useState(false)
+    const navigate = useNavigate()
+    useTitle(t('popups_add_wallet'))
+    const location = useLocation()
+    const { showSnackbar } = usePopupCustomSnackbar()
+    const { value: request } = useUnconfirmedRequest()
+
+    const wallet: Web3Plugin.ConnectionResult<ChainId, NetworkType, ProviderType> = location.state.usr
+
+    const { value: bounds } = useAsync(async () => {
+        if (!wallet.account) return false
+        return NextIDProof.queryExistedBindingByPlatform(NextIDPlatform.Ethereum, wallet.account)
+    })
+    if (bounds && bounds.length > 0 && !isBound) {
+        const res = bounds.filter((x) => x.persona === currentPersona?.publicHexKey)
+        if (res.length > 0) {
+            const final = res[0].proofs.filter((x) => {
+                return isSameAddress(x.identity, wallet?.account)
+            })
+            if (final.length > 0) setIsBound(true)
+        }
+    }
```

Comment:
Move this calculation into a `useMemo` whose returned value is `isBound`. Also, remove the `isBound` state.

Reply:
done